corehq/apps/app_manager/models.py

```diff
@@ -792,6 +792,43 @@ def short_comment(self):
         return self.comment if len(self.comment) <= 500 else self.comment[:497] + '...'
 
 
+class CaseLoadReference(DocumentSchema):
+    _allow_dynamic_properties = False
+    path = StringProperty()
+    properties = ListProperty(unicode)
+
+
+class CaseSaveReference(DocumentSchema):
+    _allow_dynamic_properties = False
+    case_type = StringProperty()
+    path = StringProperty()
+    properties = ListProperty(unicode)
+    create = BooleanProperty(default=False)
+    close = BooleanProperty(default=False)
+
+
+class CaseReferences(DocumentSchema):
+    _allow_dynamic_properties = False
+    load = DictProperty()
+    save = DictProperty()
```

Comment:
This change seems a little scary. We don't really have integration tests between HQ and Vellum so if there was a bug in vellum that caused this to not have the correct format, it wouldn't save and the user would be kind of screwed. With that said that is tested in Vellum's tests, so it shouldn't happen, but could be an extra point of failure.